Python/Product/Analysis/Values/FunctionInfo.cs

```diff
@@ -450,69 +450,75 @@ private class StringArrayComparer : IEqualityComparer<string[]> {
             }
         }
 
-        public override IEnumerable<OverloadResult> Overloads {
-            get {
-                if (_functionAttrs != null && _functionAttrs.TryGetValue("__wrapped__", out VariableDef wrapped)) {
-                    foreach (var o in wrapped.TypesNoCopy.SelectMany(n => n.Overloads)) {
-                        yield return o;
-                    }
+        public override IEnumerable<OverloadResult> Overloads => this.EnumerateOverloads(new HashSet<FunctionInfo>());
+
+        IEnumerable<OverloadResult> EnumerateOverloads(HashSet<FunctionInfo> beenThrough) {
+            if (!beenThrough.Add(this))
```

Comment:
We have `Push()`/`Pop()` functions on each value type for this kind of protection. Any reason it isn't sufficient here? (I'm totally willing to believe it's not sufficient, but I'd rather not have multiple ways of doing the same thing throughout the codebase if it's fine.)